src/Decoder.ts

```diff
@@ -82,18 +121,22 @@ const STATE_ARRAY = "array";
 const STATE_MAP_KEY = "map_key";
 const STATE_MAP_VALUE = "map_value";
 
-type MapKeyType = string | number;
+type MapKeyType = string | number | bigint | Uint8Array;
 
-const isValidMapKeyType = (key: unknown): key is MapKeyType => {
-  return typeof key === "string" || typeof key === "number";
-};
+function isValidMapKeyType(key: unknown, useMap: boolean, supportObjectNumberKeys: boolean): key is MapKeyType {
+  if (useMap) {
+    return typeof key === "string" || typeof key === "number" || typeof key === "bigint" || key instanceof Uint8Array;
```

Comment:
> Typescript doesn't offer a way to write this as a single thing like `key instanceof MapKeyType`?

Correct, this is really the only way to do it, since typescript types are not actually there at runtime.

> What is the rationale for what types are being supported? Would boolean make sense? Does the msgpack `float` family get supported, because they just become a `number`?

We (Algorand) use ints and binary data as map keys, so I wanted to expand support for those. Boolean could also make sense.

And yes msgpack floats get decoded into a JS `number`, so they are supported as map keys.